packages/react/src/composite/root/CompositeRoot.tsx

```diff
@@ -8,16 +8,22 @@ import { useRenderElement } from '../../utils/useRenderElement';
 import type { BaseUIComponentProps } from '../../utils/types';
 import type { Dimensions, ModifierKey } from '../composite';
 import { useDirection } from '../../direction-provider/DirectionContext';
-
-const COMPOSITE_ROOT_STATE = {};
+import { EMPTY_ARRAY, EMPTY_OBJECT } from '../../utils/constants';
+import { CustomStyleHookMapping } from '../../utils/getStyleHookProps';
 
 /**
  * @internal
  */
-export function CompositeRoot<Metadata extends {}>(componentProps: CompositeRoot.Props<Metadata>) {
+export function CompositeRoot<Metadata extends {}, State extends Record<string, any>>(
+  componentProps: CompositeRoot.Props<Metadata, State>,
+) {
   const {
     render,
     className,
+    refs = EMPTY_ARRAY,
+    props = EMPTY_ARRAY,
```

Comment:
`null` or `undefined` would be more efficient as default values, array `ref` makes `useRenderElement` use `useForkRefN`.

Reply:
They're spread inside the array, so it will always use `useForkRefN` regardless